app/models/event.py

```diff
@@ -29,10 +29,15 @@
 def get_new_event_identifier(length=8):
     identifier = str(binascii.b2a_hex(os.urandom(int(length / 2))), 'utf-8')
     count = get_count(Event.query.filter_by(identifier=identifier))
-    if count == 0:
-        return identifier
-    else:
+    try:
+        int(identifier)
+    except ValueError:
```

Comment:
Wrong way to check for integer and opposite logic and in wrong place.

This will only return integer identifiers

Please test before sending PRs

Write a test for it to ensure it works correctly